docs/spec_draft.md

```diff
@@ -1280,6 +1281,58 @@ For boolean element type, the behavior is same as [stablehlo.or](#stablehloor).
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.map
+
+### Semantics
+
+Performs a map function `computation` over the `inputs` along the `dimensions`
+and produces a `result` tensor where each element is the result of `computation`
+applied to the corresponding elements in `inputs`.
+
+More formally, `result[i0, ..., iR-1] = computation(inputs0[i0, ..., iR-1], `
+`..., inputsN-1[i0, ..., iR-1])`.
+
+### Inputs
+
+| Name          | Type                                             |
+|---------------|--------------------------------------------------|
+| `inputs`      | variadic number of tensors of any supported type |
+| `dimensions`  | 1-dimensional tensor constant of type `si64`     |
+| `computation` | `function`                                       |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `inputs` and `result` have the same shape.
+  * (C2) size(`inputs`) $=$ N $\ge$ 1.
+  * (C3) size(`dimensions`) $=$ rank(`inputs[k]`) for all `k` $\in$ [0, N).
+  * (C4) `dimensions[i]` $\lt$ `dimensions[i+1]` for all `i` $\in$ [0, `R`-2].
+  * (C5) `computation` has type `(tensor<E0>, ..., tensor<EN-1>) -> tensor<E'>`
+    where `Ek` $=$ element_type(`inputs[k]`) and `E'` $=$
+    element_type(`result`).
```

Comment:
What does "computation of type T_0, T_1,..., T_{N + M -1} -> S with N parameters of type T and M of arbitrary type" mean in https://www.tensorflow.org/xla/operation_semantics#map? Remnants of the project to generalize `dimensions`?

Reply:
It's unrelated to `dimensions`. "M" is a type that seems to be deprecated now, and the documentation has some artifacts of the change still left over.